### PR TITLE
Setting __event_name for unhandled errors (SCP-4279)

### PR DESCRIPTION
**Pull request by devonbush**
Fixes a bug where `__event_name` was read in the `exit_pipeline` function, but not set if the global exception handler was triggered.

### CODE REVIEW

File: ingest/ingest_pipeline.py [hunk 1]
```diff
@@ -576,29 +576,35 @@ def main() -> None:
...
     """
+    parsed_args = create_parser().parse_args()
+    validate_arguments(parsed_args)
+    arguments = vars(parsed_args)
     status = 0
+    status_cell_metadata = 0
+    ingest = None
+
+    # Initialize global variables for current ingest job
+    config.init(
```
Comment (jlchang) [inline ingest/ingest_pipeline.py:587]: local testing shows this erroring out:
```
Traceback (most recent call last):  
  File "ingest_pipeline.py", line 614, in <module>  
    main()  
  File "ingest_pipeline.py", line 608, in main  
    MetricsService.log(config.get_parent_event_name(), config.get_metric_properties())  
  File "/Users/jlchang/Documents/GitHub/scp-ingest-pipeline/ingest/config.py", line 28, in get_parent_event_name  
    return __event_name  
NameError: name '__event_name' is not defined
```
Moving `try:` down one line so the IngestPipeline object is instantiated should resolve this error.

**Review by jlchang — CHANGES_REQUESTED** [review]
Having `try:` above IngestPipeline object instantiation results in `NameError: name '__event_name' is not defined` because config isn't fully initialized ([more details](https://broadinstitute.slack.com/archives/CBEHTH601/p1650115317813179?thread_ts=1650107131.342889&cid=CBEHTH601))

Shifting the `try:` statement down one line should resolve the NameError.

Not entirely clear if it will yield the logging (Sentry or log files) we're hoping for, but it should get us closer...

**Review by eweitz — APPROVED** [review]
Could you add a PR description, for some context?  

To my understanding, the intent here is to improve observability of unhandled errors in Mixpanel -- something like that seems like a more useful PR title, to help us recall what this PR did after it's no longer top-of-mind.

That said, code looks good!

**Review by jlchang — APPROVED** [review]
Confirmed that simulated error will result in stack trace to `log.txt`
confirmed Mixpanel logging of `ingest-pipeline:unhandled-exception:ingest` events

**Pull request merged**